### PR TITLE
Enforce type in custom screen definition (#9930)

### LINKED ISSUE #9930: Enforce type in custom screen definition
When registering a custom screen , user need to define the type of custom screen (filed "type") , for example  : 
 
```
   const customScreenExample = {
        id: 'testId',
        name: 'testName',
        type: 'CARD_LIST',
        initialBusinessPeriod: 'FROM_TODAY_TO_YEAR_END',
        processIds: [
            'api_test',
            'cypress',
            'defaultProcess',
```

Today it is possible to load the custom screen without this value , but in this case the bubble display does not work on custom tiles in the dashboard

The issue is to enforce the setting of this value . 

If the value is not set , the customscreen shall not be loaded and an error should appear on the console  

Code should be add in CustomScreenService.ts

Add unit test 

Add in migration guide for next release 4.13

### PR DESCRIPTION
**Pull request by freddidierRTE**
- In release notes :
  -  In chapter : Features 
  -  Text : `#9930  Enforce type in custom screen definition `

### CODE REVIEW

File: frontend/src/app/services/customScreen/CustomScreenService.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        );
+    });
+
+    it('should not add a custom screen definition when type is invalid and log an error', () => {
+        const errorSpy = spyOn(LoggerService, 'error');
+        const screenDef = {
+            id: 'screen3',
+            name: 'Screen Three',
+            type: 'INVALID_TYPE'
+        } as unknown as ScreenDefinition;
```
Comment (vlo-rte) [inline frontend/src/app/services/customScreen/CustomScreenService.spec.ts]: What do you think about this solution : 

const invalidScreenDef: any = {
    id: 'screen3',
    name: 'Screen Three',
    type: 'INVALID_TYPE'
};

It would avoid the double cast.

**Review by vlo-rte — COMMENTED** [review]
Maybe we could add a test to verify the behavior in case of duplicate screen ids (to avoid future regressions) ?

**Review by vlo-rte — APPROVED** [review]
